server/server.js

```diff
@@ -86,7 +86,8 @@ UserPresence = {
 	},
 
 	createConnection: function(userId, connection, status, metadata) {
-		if (!userId || !connection.id) {
+		// if connections is invalid, does not have an userId or is already closed, don't save it on db
+		if (!userId || !connection.id || connection.closed) {
```

Comment:
I tried this approach but sometimes connection was created because `connection.closed` was equal `true` after `if` statement was executed. I also tried this:
```javascript
if (!connection.closed) {
      UsersSessions.upsert(query, update);
}
```
Also sometimes connection was created.

Reply:
I couldn't get this extreme race condition. from my tests the connection was never created.

another solution then would be always call `UserPresence.removeConnection` from `onClose` callback, instead of checking if `connection.UserPresenceUserId` exists.

Reply:
I checked that `onClose` callback are called before:
```javascript
Accounts.onLogin(function(login) {
     UserPresence.createConnection(login.user._id, login.connection);
});
```
So if you remove ``if (connection.UserPresenceUserId !== undefined && connection.UserPresenceUserId !== null)`` you will get same effect.

Reply:
I tested this code a lot today and it didn't create a connection. as you said `onClose` is always called before `createdConnection`, that's why it works.

there is one situation I still get zombie connections when the client is forced killed.. in this case `onClose` is called.

Reply:
I using Realtime API.

Reply:
@sampaiodiego why you don't like my solution? I think it don't have any performance drawbacks because connection are removed only if `connection.closed` equals `true`?

Reply:
what doesn't feel right to me is that it is inserting something on db and removing it on the line bellow. it doesn't add much load on db because it doesn't happen so often but still are two operations on db where none is required.

Reply:
@distlibs I've added the following test to make sure it will not insert if connection already marked as closed, wdyt? https://github.com/Konecty/meteor-user-presence/pull/31/commits/212c4edfb107dd754a95e3828ff5459daacce81e

Reply:
Ok.